src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -11089,7 +11089,7 @@ instruct convF2D_reg_mem(regD dst, memory src)
 instruct convD2F_reg_reg(regF dst, regD src)
 %{
   match(Set dst (ConvD2F src));
-
+  effect(TEMP dst);
```

Comment:
You don't need `TEMP dst`, if `dst` is an alias of `src` then a destructive `xor` is not emitted.

Reply:
Without TEMP annotation dst and src may be aliased if src live range does not survives beyond this instruction.

Reply:
I had checked before submitting the PR, the cvt  xmm0, xmm0, xmm0 form was slower than xorps xmm1, xmm1 followed by cvt xmm1, xmm1, xmm0.